bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -628,6 +690,22 @@ public long addEntry(byte[] data) throws InterruptedException, BKException {
         return addEntry(data, 0, data.length);
     }
 
+    @Override
+    public CompletableFuture<Long> append(ByteBuffer data) {
```

Comment:
you made this class (LedgerHandle.java) implement client.api.WriteHandle, this is not making sense to me 

1) if your plan is to make org.apache.bookkeeper.client.api public package and org.apache.bookkeeper.client internal package, why do you want internal package class implements external interface
2) ReadOnlyLedgerHandle extends LedgerHandle class, so these methods are also methods of ReadOnlyLedgerHandle, but that doesn't make sense for ReadOnlyLedgerHandle to have 'append' methods
3) what is the relation between client package handles (ReadOnlyLedgerHandle and LedgerHandle) and client.api package handles (Handle,  ReadHandle, WriteHandle)

Reply:
client.api is a totally new API, clients will never see the old messy API. ReadOnlyLedgerHandle extends LedgerHandle and this does not make sense, it has addEntry methods for instance.
I think we should leave the old API as it is now and work on a new clean API (see comments above)

Reply:
@reddycharan the way to think about this - `client.api` will be new api, `BookKeeper` class will become the implementation class. the old methods in `BookKeeper` will be deprecated at 4.7.